docs/DeveloperGuide.md

```diff
@@ -254,74 +254,353 @@ _{Explain here how the data archiving feature will be implemented}_
+
+*  2b. TutorAid detects an invalid class index number (e.g. a negative index number, a non-integer index number, a non-existent index number)
+   
+    *  2b1. TutorAid displays an error message highlighting the invlaid class index number and requests the tutor to re-enter the command.
+       
+    *  2b2. User re-enters the command with another class index number.
+
+        Steps 2b1-2b2 are repeated until the command entered is correct.
+
+        Use case resumes from step 3.
+
+*  2c. TutorAid detects an invalid date (e.g. a wrongly formatted date, non-existent date, day is less than 1 and more than 31, month is less than 1 and more than 12, year is less than the current year)
+   
+    *  2c1. TutorAid displays an error message highlighting the invalid input for the date and requests the tutor to re-enter the command.
+       
+    *  2c2. User re-enters the command with another date.
+
+        Steps 2c1-2c2 are repeated until the command entered is correct.
+
+        Use case resumes from step 3.
+
+*  2d. TutorAid detects an invalid start/end timings (e.g. wrongly formatted timing, hours stated is less than 0 or more than 23, minutes stated is less than 0 or more than 59)
+   
+    *  2d1. TutorAid displays an error message highlighting the invalid timings and requests the tutor to re-enter the command.
+       
+    *  2d2. User re-enters the command with another pair of timings.
+
+        Steps 2d1-2d2 are repeated until the command entered is correct.
+
+        Use case resumes from step 3.
+
+*  2e. TutorAid detects a clash with another existing class (e.g. overlapping timings on the same date)
+   
+    *  2e1. TutorAid displays an error message highlighting the clash in classes and requests the tutor to key a different day and/or pair of timings.
+       
+    *  2e2. User re-enters the command with A different day and/or different pair of timings.
+        
+        Steps 2e1-2e2 are repeated until the command entered is correct.
+
+        Use case resumes from step 3.
+
+**Use Case 18: Edit pricing of a class**
+
+**MSS**
+1. Tutor identifies the class he/she wishes to modify and takes note of the index number.
+2. Tutor keys in the command to modify the pricing of a class, along with the new price of the class and the class index number.
+3. TutorAid successfully updates the pricing of the class corresponding to the given class index number.
+Use case ends.
+
+**Extensions**
+*  1a. TutorAid detects an error in the command to list classes (e.g. command is misspelt)
+   
+    *  1a1. TutorAid displays an error message highlighting the wrongly formatted command and requests the tutor to re-enter the command.
+       
+    *  1a2. User re-enters the command.
+
+        Steps 1a1-1a2 are repeated until the data entered are correct.
+
+        Use case resumes from step 2.
+    
+*  2a. TutorAid detects a wrongly formatted command to edit the pricing of a class (e.g. missing fields in the component, typos in the command)
```

Comment:
Do you think if it is okay for us to put all the invalid input together as one extension? At least if the input comes from the same step in the MSS